### DISCUSSION TITLE
AssemblyVersion default value set by MinVer

### DISCUSSION THREAD

**Discussion by ghost** [adamralph/minver#856] [Q&A] [answered]
Just for my curiosity and because I couldn't find anything about this topic in the documentation, why the default value of `AssemblyVersion` set by MinVer is `$(MinVerMajor).0.0.0` instead of `$(MinVerMajor).$(MinVerMinor).$(MinVerPatch).0` as for `FileVersion`?

https://github.com/adamralph/minver/blob/c0206b5421208b591cde2a42cfa439451f3af8d4/MinVer/build/MinVer.targets#L64-L65

According to the Microsoft documentation (https://learn.microsoft.com/en-us/dotnet/standard/assembly/versioning#assembly-version-number), there is no reason not to set the `AssemblyVersion` to `$(MinVerMajor).$(MinVerMinor).$(MinVerPatch).0`

**Comment by adamralph** [comment, marked as answer]
It's in the documentation. Please see [_Version numbers_](https://github.com/adamralph/minver#version-numbers):

> Those properties are used to set the following .NET SDK properties, satisfying the official [open-source library guidance for version numbers](https://docs.microsoft.com/en-ca/dotnet/standard/library-guidance/versioning#version-numbers):
> 
> | Property          | Value                                         |
> | ----------------- | --------------------------------------------- |
> | `AssemblyVersion` | `{MinVerMajor}.0.0.0`                         |
> | `FileVersion`     | `{MinVerMajor}.{MinVerMinor}.{MinVerPatch}.0` |
> | `PackageVersion`  | `{MinVerVersion}`                             |
> | `Version`         | `{MinVerVersion}`                             |

And following that link:

> ✔️ CONSIDER only including a major version in the AssemblyVersion.
> 
> > e.g. Library 1.0 and Library 1.0.1 both have an AssemblyVersion of `1.0.0.0`, while Library 2.0 has AssemblyVersion of `2.0.0.0`. When the assembly version changes less often, it reduces binding redirects.

**Reply by ghost** [reply]
I was not reading the right documentation! Thank you!
